pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -235,6 +236,15 @@ public void deleteTable(String tableNameWithType)
       throws Exception {
     // Wait externalview to converge
     long endTimeMs = System.currentTimeMillis() + _externalViewDroppedMaxWaitMs;
+    _tableDataManagerMap.compute(tableNameWithType, (k, v) -> {
```

Comment:
Don't use `compute()` here. `get()` should be enough. Currently this compute will remove the table data manager, preventing it being shut down when external view is removed